platform-api/che-core-api-infrastructure-local/src/main/java/org/eclipse/che/api/local/LocalStackDaoImpl.java

```diff
@@ -0,0 +1,197 @@
+import org.eclipse.che.api.machine.server.recipe.adapters.PermissionsSerializer;
+import org.eclipse.che.api.machine.shared.Group;
+import org.eclipse.che.api.machine.shared.Permissions;
+import org.eclipse.che.api.workspace.server.model.impl.stack.DecoratedStackImpl;
+import org.eclipse.che.api.workspace.server.model.impl.stack.Stack;
+import org.eclipse.che.api.core.model.workspace.EnvironmentState;
+import org.eclipse.che.api.core.model.workspace.ProjectConfig;
+import org.eclipse.che.api.core.model.workspace.WorkspaceConfig;
+import org.eclipse.che.api.local.storage.LocalStorage;
+import org.eclipse.che.api.local.storage.LocalStorageFactory;
+import org.eclipse.che.api.workspace.server.model.impl.stack.StackComponent;
+import org.eclipse.che.api.workspace.server.dao.StackDao;
+import org.eclipse.che.api.workspace.server.model.impl.stack.StackSource;
+import org.eclipse.che.api.workspace.server.stack.adapters.ByteArrayToBase64Serializer;
+import org.eclipse.che.api.workspace.server.stack.adapters.CommandSerializer;
+import org.eclipse.che.api.workspace.server.stack.adapters.EnvironmentStateSerializer;
+import org.eclipse.che.api.workspace.server.stack.adapters.LimitsSerializer;
+import org.eclipse.che.api.workspace.server.stack.adapters.MachineConfigSerializer;
+import org.eclipse.che.api.workspace.server.stack.adapters.MachineSourceSerializer;
+import org.eclipse.che.api.workspace.server.stack.adapters.ProjectConfigSerializer;
+import org.eclipse.che.api.workspace.server.stack.adapters.RecipeSerializer;
+import org.eclipse.che.api.workspace.server.stack.adapters.StackComponentSerializer;
+import org.eclipse.che.api.workspace.server.stack.adapters.StackSourceSerializer;
+import org.eclipse.che.api.workspace.server.stack.adapters.WorkspaceConfigSerializer;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
+import java.io.IOException;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * Implementation local storage for {@link Stack}
+ *
+ * @author Alexander Andrienko
+ */
+@Singleton
+public class LocalStackDaoImpl implements StackDao {
+
+    private static final String STORAGE_FILE = "stacks.json";
+
+    private LocalStorage                stackStorage;
+    private Map<String, DecoratedStackImpl> stacks;
+    private ReadWriteLock               lock;
+
+    @Inject
+    public LocalStackDaoImpl(LocalStorageFactory localStorageFactory) throws IOException, ServerException {
+        HashMap<Class<?>, Object> adapters = new HashMap<>();
+        adapters.put(StackComponent.class, new StackComponentSerializer());
```

Comment:
As far as i can see it in rather `Adapter` than `Serializer` please change the names of these classes.